src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1263,6 +1405,81 @@ protected virtual async Task<DateTime> OnRenewLockAsync(string lockToken)
             return lockedUntilUtc;
         }
 
+        protected virtual async Task<DateTime> OnRenewLocksAsync(IEnumerable<string> lockTokens)
+        {
+            DateTime lockedUntilUtc;
+            try
+            {
+                // Create an AmqpRequest Message to renew locks
+                var amqpRequestMessage = AmqpRequestMessage.CreateRequest(ManagementConstants.Operations.RenewLockOperation, this.OperationTimeout, null);
+
+                if (this.ReceiveLinkManager.TryGetOpenedObject(out var receiveLink))
+                {
+                    amqpRequestMessage.AmqpMessage.ApplicationProperties.Map[ManagementConstants.Request.AssociatedLinkName] = receiveLink.Name;
+                }
+                amqpRequestMessage.Map[ManagementConstants.Properties.LockTokens] = new[] { lockTokens.Select(lockToken => new Guid(lockToken)) };
+
+                var amqpResponseMessage = await this.ExecuteRequestResponseAsync(amqpRequestMessage).ConfigureAwait(false);
+
+                if (amqpResponseMessage.StatusCode == AmqpResponseStatusCode.OK)
+                {
+                    var lockedUntilUtcTimes = amqpResponseMessage.GetValue<IEnumerable<DateTime>>(ManagementConstants.Properties.Expirations);
+                    lockedUntilUtc = lockedUntilUtcTimes.Last();
```

Comment:
Given there are multiple messages in the batch, there could should it be the last or the first renewed message's date/time?
Or perhaps the earliest expiring date/time?